nixos/modules/services/desktops/gnome/gcr-ssh-agent.nix

```diff
@@ -0,0 +1,32 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+let
+  cfg = config.services.gnome.gcr-ssh-agent;
+in
+{
+  meta = {
+    maintainers = lib.teams.gnome.members;
+  };
+
+  options = {
+    services.gnome.gcr-ssh-agent = {
+      enable = lib.mkEnableOption "GCR ssh-agent";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    systemd.packages = [ pkgs.gcr_4 ];
+    systemd.user.sockets.gcr-ssh-agent.wantedBy = [ "sockets.target" ];
```

Comment:
> Also, systemd.user.sockets.gcr-ssh-agent.wantedBy = [ "sockets.target" ] was missing in the original service definition. I'll add it too

Why is that needed? Reading <https://www.freedesktop.org/software/systemd/man/latest/systemd.special.html#sockets.target>, I do not think it applies:

- It cannot start after boot since it is a user service.
- It cannot rely on socket activation because by the time something could do it, we would already need to have a session the something could run in, and then it would be too late for the environment to be updated. That’s why the service is `WantedBy=default.target`, I believe.

Reply:
That does make sense. I'll remove it, I have just seen it on other socket services, but I did not read the `freedesktop` documentation for it.

Reply:
> It cannot start after boot since it is a user service.

This code is setting the socket to be enabled for the user manager by default, it has nothing to do with boot

> It cannot rely on socket activation because by the time something could do it, we would already need to have a session the something could run in, and then it would be too late for the environment to be updated.

It relies on socket activation just fine. If you only enable the service and not the socket, then you don't get environment updating _at all_ since that is only done as an `ExecStartPost` on the .socket unit (though I have seen noise upstream about stopping setting it from systemd units by default and use some other mechanism)